File: backend/app/schemas/platform_updates.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum


class UpdateType(str, Enum):
    FEATURE = "FEATURE"
    BUG_FIX = "BUG_FIX"
    ANNOUNCEMENT = "ANNOUNCEMENT"
    MAINTENANCE = "MAINTENANCE"
    SECURITY = "SECURITY"


class UpdatePriority(str, Enum):
    LOW = "LOW"
    NORMAL = "NORMAL"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"


class TargetAudience(str, Enum):
    ALL = "ALL"
    ADMIN = "ADMIN"
    USER = "USER"


class PlatformUpdateBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: str = Field(..., min_length=1)
    update_type: UpdateType = UpdateType.ANNOUNCEMENT
    priority: UpdatePriority = UpdatePriority.NORMAL
    media_type: Optional[str] = Field(None, max_length=20)  # IMAGE, VIDEO, NONE
    media_urls: Optional[List[str]] = None
    thumbnail_url: Optional[str] = Field(None, max_length=500)
    target_audience: TargetAudience = TargetAudience.ALL


class PlatformUpdateCreate(PlatformUpdateBase):
    pass


class PlatformUpdateUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = Field(None, min_length=1)
    update_type: Optional[UpdateType] = None
    priority: Optional[UpdatePriority] = None
    media_type: Optional[str] = Field(None, max_length=20)
    media_urls: Optional[List[str]] = None
    thumbnail_url: Optional[str] = Field(None, max_length=500)
    target_audience: Optional[TargetAudience] = None
    is_published: Optional[bool] = None
    published_at: Optional[datetime] = None
    expires_at: Optional[datetime] = None


class PlatformUpdateResponse(PlatformUpdateBase):
    id: int
    is_published: bool
    view_count: int
    published_at: Optional[datetime] = None
    expires_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class PlatformUpdateList(BaseModel):
    items: List[PlatformUpdateResponse]
    total: int
    page: int
    page_size: int


class UserUpdateViewResponse(BaseModel):
    id: int
    update_id: int
    user_id: int
    viewed_at: datetime
    
    class Config:
        from_attributes = True


class UserUpdateViewList(BaseModel):
    items: List[UserUpdateViewResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/platformUpdates.ts
/**
 * Platform Updates Types
 * Based on backend/app/schemas/platform_updates.py
 */

// Update Type Enum
export enum UpdateType {
  FEATURE = "FEATURE",
  BUG_FIX = "BUG_FIX",
  ANNOUNCEMENT = "ANNOUNCEMENT",
  MAINTENANCE = "MAINTENANCE",
  SECURITY = "SECURITY",
}

// Update Priority Enum
export enum UpdatePriority {
  LOW = "LOW",
  NORMAL = "NORMAL",
  HIGH = "HIGH",
  CRITICAL = "CRITICAL",
}

// Target Audience Enum
export enum TargetAudience {
  ALL = "ALL",
  ADMIN = "ADMIN",
  USER = "USER",
}

// Platform Update Base Type
export interface PlatformUpdateBase {
  title: string;
  description: string;
  update_type: UpdateType;
  priority: UpdatePriority;
  media_type?: string;
  media_urls?: string[];
  thumbnail_url?: string;
  target_audience: TargetAudience;
}

// Platform Update Create Type
export type PlatformUpdateCreate = PlatformUpdateBase;

// Platform Update Update Type
export interface PlatformUpdateUpdate {
  title?: string;
  description?: string;
  update_type?: UpdateType;
  priority?: UpdatePriority;
  media_type?: string;
  media_urls?: string[];
  thumbnail_url?: string;
  target_audience?: TargetAudience;
  is_published?: boolean;
  published_at?: string | null;
  expires_at?: string | null;
}

// Platform Update Response Type
export interface PlatformUpdateResponse extends PlatformUpdateBase {
  id: number;
  is_published: boolean;
  view_count: number;
  published_at?: string | null;
  expires_at?: string | null;
  created_at: string;
  updated_at: string;
}

// Platform Update List Type (paginated results)
export interface PlatformUpdateList {
  items: PlatformUpdateResponse[];
  total: number;
  page: number;
  page_size: number;
}

// User Update View Response Type
export interface UserUpdateViewResponse {
  id: number;
  update_id: number;
  user_id: number;
  viewed_at: string;
}

// User Update View List Type (paginated results)
export interface UserUpdateViewList {
  items: UserUpdateViewResponse[];
  total: number;
  page: number;
  page_size: number;
}

// Export all types
export type PlatformUpdate = PlatformUpdateResponse;
